presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java

```diff
@@ -291,7 +291,7 @@ public void addStateChangeListener(StateChangeListener<QueryState> stateChangeLi
         try {
             return tryGetFutureValue(queryExecutionFuture);
         }
-        catch (Exception ignored) {
+        catch (Throwable ignored) {
```

Comment:
Catching `Error`s without discerning is a bad practice unless there is really no alternative.
If `StackOverflowError` is possible, it would be better to handle it explicitly (as we do in some cases).

Based on your comment, I understand it's not currently possible to get StackOverflowError here, so i suggest we don't add this.

Reply:
I've come to the opinion that `tryGetFutureValue` and `getFutureValue` methods in `MoreFutures` are generally anti patterns. See the discussion here: https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate#encourages-unwrapping-exceptions-from-other-threads

If we remove usage of  `tryGetFutureValue`, then we don't have to worry about `Error`.

Reply:
Thanks for the comment and I understood,